mibidata/tiff.py

```diff
@@ -82,12 +84,17 @@ def write(filename, image, sed=None, optical=None, ranges=None,
         raise ValueError('Image coordinates and size must not be None.')
     if image.masses is None or image.targets is None:
         raise ValueError('Image channels must contain both masses and targets.')
-    if np.issubdtype(image.data.dtype, np.integer) and not write_float:
-        range_dtype = 'I'
+    if write_dtype and not write_dtype in ['float', 'int', np.float, np.uint16]:
+        raise ValueError('Invalid dtype specification.')
+    if not write_dtype:
+        range_dtype = 'I' if\
+            np.issubdtype(image.data.dtype, np.integer) else 'd'
     else:
-        range_dtype = 'd'
+        range_dtype = 'I' if write_dtype in ['int', np.uint16] else 'd'
+
```

Comment:
If you end up doing this, you're not respecting the type-choices you promised before. Especially for int types, you pre-define 'int' and np.uint16 as accepted types, but you end up using int. Similar with float.

Reply:
This is for a specific TIFF tag, not the data itself, and needs to conform to what's allowed for that tag.